### PR TITLE
Add s3 support for Announcements model

### LINKED ISSUE #623: Add S3 support for Announcements Model
### Subject of the issue
- Announcement model has an image and file field. All these data should be stored in S3 bucket.

### Expected behaviour
- Announcement list page should link images and uploads to S3 URL.

### PR DESCRIPTION
**Pull request by kracekumar**
### Issue Reference
This PR addresses the Issue : Fixes #623 

### Summarize
Please, describe what the PR does, the changes you have made.

- PR adds Announcement model with s3 support.
- To test create a s3 account, with bucket, and a folder inside the bucket named `media`.
- Give public read access to the bucket.
- Copy .env.example to .env and replace s3 bucket names, access key, secret key.

> Also, mention the key points if any to look into while code reviews

- Please check migrations file.

@biswaz Please review this.

### CODE REVIEW

**Comment by riginoommen** [comment]
Hello @kracekumar 

Thanks for the merge request can you please squash the multiple commits into one and assign the issue number that you worked on

https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

File: mainapp/models.py [hunk 1]
```diff
@@ -342,21 +349,20 @@ def __str__(self):
...
-        verbose_name='Type'
-    )
+    priority = models.CharField(
+        max_length=20,
+        choices = announcement_priorities,
+        verbose_name='Priority',
+        default='L')
+
+    description = models.TextField(blank=True)
+    image = models.ImageField(blank=True, null=True, upload_to='media')
```
Comment (ragsagar) [inline mainapp/models.py]: Don't use `null=True`, use just `blank=True`. `ImageField` and `FileField` are `CharField` internally.

**Comment by kracekumar** [comment]
@ragsagar Fixed.
@riginoommen @biswaz Rebased.

**Comment by akshayjoyinfo** [comment]
There are couple of issues in this PR.

1. When we make pinned announcements 
   Then the pinned announcement won't shown in the list ( but shows in Red color as priority, but no hyperlink for anouncements
s3 aaaa

2. What if the annoucement team uploads file (with same name, chances are there) in that s3 will throw exception, 

image


@kracekumar , @biswaz , @bobinson  -- go through my comments

**Comment by kracekumar** [comment]
> When we make pinned announcements
Then the pinned announcement won't shown in the list ( but shows in Red color as priority, but no hyperlink for anouncements

Yes, pinned announcements aren't displayed in the table. No individual URL for the announcement. This by design. It's easy to add detail view for an announcement. /cc @naveenpf 

> What if the annoucement team uploads file (with same name, chances are there) in that s3 will throw exception,

Good catch. I'll have to create a unique filename for every upload.

**Pull request merged**